File: Practica2_i-Lunch/src/gestorAplicacion/gestionRestaurante/producto.py
class Producto:

    # Atributos estaticos

    _productos = []

    # Constructor

    def __init__(self, nombre = "", descripcion = "", precio = 0, disponibilidad = False, restriccion = False, cantidad = 0):
        self._nombre = nombre
        self._descripcion = descripcion
        self._precio = precio
        self._restriccion = restriccion
        self._disponibilidad= disponibilidad
        self._cantidad = cantidad
        
        self._estado = False # Sin preparar aun

        Producto._productos.append(self)

    # Getters y Setters

    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre

    def getDescripcion(self):
        return self._descripcion

    def setDescripcion(self, descripcion):
        self._descripcion = descripcion

    def getPrecio(self):
        return self._precio

    def setPrecio(self, precio):
        self._precio = precio

    def getDisponibilidad(self):
        return self._disponibilidad

    def setDisponibilidad(self, disponibilidad):
        self._disponibilidad = disponibilidad

    def getRestriccion(self):
        return self._restriccion

    def setRestriccion(self, restriccion):
        self._restriccion =  restriccion

    def getCantidad(self):
        return self._cantidad

    def setCantidad(self, cantidad):
        self._cantidad = cantidad

    def getEstado(self):
        return self._estado

    def setEstado(self, estado):
        self._estado = estado


    @classmethod
    def getProductos(cls):
        return cls._productos

    @classmethod
    def setProductos(cls, productos):
        cls._productos = productos

    # ToString

    def __str__(self):
        return f"Nombre: {self._nombre} \n" \
               f"Descripcion: {self._descripcion} \n" \
               f"Precio: ${self._precio}\n" \
               f"Disponibilidad: {self._disponibilidad} \n" \
               f"Restriccion de edad: {self._restriccion}\n" \
               f"Cantidad disponible: {self._cantidad}"
    

File: Practica1_i-Lunch/src/gestorAplicacion/gestionRestaurante/Producto.java
package gestorAplicacion.gestionRestaurante;

import java.io.Serializable;
import java.util.ArrayList;

public class Producto implements Serializable {

	// Serializcion
	private static final long serialVersionUID = 1L;
	private static ArrayList<Producto> productos;
	static {
		productos = new ArrayList<Producto>();
	}

	// Atributos
	private String nombre;
	private String descripcion;
	private int precio;
	private boolean disponiblidad;
	private boolean restriccion;
	private int cantidad;
	private boolean estado;

	// Constructores
	public Producto(String nombre, String descripcion, int precio, boolean disponiblidad, boolean restriccion,
			int cantidad) {
		super();
		this.nombre = nombre;
		this.descripcion = descripcion;
		this.precio = precio;
		this.disponiblidad = disponiblidad;
		this.restriccion = restriccion;
		this.cantidad = cantidad;
		this.estado = false; /* <-- false significa sin preparar y true, preparado */
		productos.add(this);
	}

	public Producto() {
		this("", "", 0, false, false, 0);
	}

	// Getters y Setters
	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getDescripcion() {
		return descripcion;
	}

	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	public int getPrecio() {
		return precio;
	}

	public void setPrecio(int precio) {
		this.precio = precio;
	}

	public boolean getDisponiblidad() {
		return disponiblidad;
	}

	public void setDisponiblidad(boolean disponiblidad) {
		this.disponiblidad = disponiblidad;
	}

	public boolean getRestriccion() {
		return restriccion;
	}

	public void setRestriccion(boolean restriccion) {
		this.restriccion = restriccion;
	}

	public int getCantidad() {
		return cantidad;
	}

	public void setCantidad(int cantidad) {
		this.cantidad = cantidad;
	}

	public boolean getEstado() {
		return estado;
	}

	public void setEstado(boolean estado) {
		this.estado = estado;
	}

	public static ArrayList<Producto> getProductos() {
		return productos;
	}

	@Override
	public String toString() {
		return "Nombre: \"" + nombre + "\"\n" + "Descripcion: \"" + descripcion + "\"\n" + "Precio: $" + precio + "\n"
				+ "Disponiblidad: " + disponiblidad + "\n" + "Restriccion de edad: " + restriccion + "\n"
				+ "Cantidad disponible: " + cantidad;
	}
}
